exchanges/okcoin/okcoin.go

```diff
@@ -12,613 +12,1658 @@ import (
+func (o *Okcoin) AccountBalanceTransfer(ctx context.Context, arg *IntraAccountTransferParam) (*SubAccountTransferResponse, error) {
+	if arg == nil {
+		return nil, common.ErrNilPointer
+	}
+	if arg.Ccy == "" {
+		return nil, currency.ErrCurrencyCodeEmpty
+	}
+	if arg.Amount <= 0 {
+		return nil, fmt.Errorf("%w amount: %f", errInvalidAmount, arg.Amount)
+	}
+	// 6:Funding Account 18:Trading account
+	if arg.From != "6" && arg.From != "18" {
+		return nil, fmt.Errorf("invalid source account type %s, 6:Funding Account 18:Trading account", arg.From)
+	}
+	// 6:Funding Account 18:Trading account
+	if arg.To != "6" && arg.To != "18" {
+		return nil, fmt.Errorf("invalid destination account type %s, 6:Funding Account 18:Trading account", arg.To)
+	}
+	if arg.FromSubAccount == "" {
+		return nil, fmt.Errorf("%w, source subaccount must be specified", errSubAccountNameRequired)
+	}
+	if arg.ToSubAccount == "" {
+		return nil, fmt.Errorf("%w, destination subaccount must be specified", errSubAccountNameRequired)
+	}
+	var resp []SubAccountTransferResponse
+	err := o.SendHTTPRequest(ctx, exchange.RestSpot, masterAccountsManageTransfersBetweenSubaccountEPL, http.MethodPost, typeAssets, "subaccount/transfer", arg, &resp, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 0 {
+		return nil, errNoValidResponseFromServer
+	}
+	return &resp[0], nil
 }
 
-// PlaceMultipleMarginOrders Place multiple orders for specific trading pairs (up to 4 trading pairs, maximum 4 orders each)
-func (o *OKCoin) PlaceMultipleMarginOrders(ctx context.Context, request []PlaceOrderRequest) (map[string][]PlaceOrderResponse, []error) {
-	currencyPairOrders := make(map[string]int)
-	resp := make(map[string][]PlaceOrderResponse)
-	for i := range request {
-		currencyPairOrders[request[i].InstrumentID]++
+// --------------------- Trade endpoints ---------------------------
+
+func (a *PlaceTradeOrderParam) validateTradeOrderParameter() error {
+	if a == nil {
+		return errNilArgument
 	}
-	if len(currencyPairOrders) > 4 {
-		return resp, []error{errors.New("up to 4 trading pairs")}
+	if a.InstrumentID.IsEmpty() {
+		return fmt.Errorf("%w, instrument id is required", currency.ErrCurrencyPairEmpty)
 	}
-	for _, orderCount := range currencyPairOrders {
-		if orderCount > 4 {
-			return resp, []error{errors.New("maximum 4 orders for each pair")}
-		}
+	if a.TradeMode == "" {
+		return errTradeModeIsRequired
 	}
+	if a.Side == "" {
+		return fmt.Errorf("%w, empty order side", order.ErrSideIsInvalid)
+	}
+	if a.OrderType == "" {
+		return fmt.Errorf("%w, empty order type", order.ErrTypeIsInvalid)
+	}
+	if a.Size <= 0 {
+		return fmt.Errorf("%w, size: %f", errInvalidAmount, a.Size)
+	}
+	return nil
+}
 
-	err := o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, marginTradingSubsection, batchOrders, request, &resp, true)
+// PlaceOrder to place a trade order.
+func (o *Okcoin) PlaceOrder(ctx context.Context, arg *PlaceTradeOrderParam) (*TradeOrderResponse, error) {
```

Comment:
Client order id is optional:
https://www.okcoin.com/docs-v5/en/#rest-api-trade-place-order
Use omitempty